src/client/public/scripts/components/nav-element.ts

```diff
@@ -6,20 +6,29 @@ import {BaseElement, property} from './base-element.js';
 
 export class NavElement extends BaseElement {
   @property() data: api.UserResponse|null = null;
+  @property() documentUrl = document.location.pathname;
 
   async connectedCallback() {
     const response = await fetch('/api/user', {credentials: 'include'});
     // TODO: change this to use the JSON API response
     this.data = (await response.json()).data as api.UserResponse;
+
+    document.body.addEventListener('url-changed', this.urlChanged.bind(this));
+  }
+
+  private urlChanged() {
+    this.documentUrl = document.location.pathname;
   }
 
   private userTemplate(): TemplateResult {
     if (!this.data) {
       return html``;
     }
 
+    const selected = this.documentUrl === '/' ? 'selected' : '';
```

Comment:
The visible page is set based on a "prefix" value. These selected items are set based on URL.

Could we instead have a single prefix value that the app-element uses to show a visible page and then give the value to the nav-element as a property / attribute. From this, can we render the elements and then query for an element with attribute `document.querySelector('*[prefix=${selectedPrefix}]')`. Then just set selected on that element.

Reply:
I think we synced offline on this.